### DISCUSSION TITLE
Question about how to detach from controlling terminal

### DISCUSSION THREAD

**Discussion by unixapple** [tauri-apps/tauri#5400] [Q&A] [answered]
Hi,

Since my application need sudo privilege, it can only now only be launched from Linux through the terminal, where I type commands like:

sudo xxx.appImage

however, this tauri application is attached to the terminal and when the terminal is closed, the application is closed too.

I want to know how to make tauri detach from the controlling terminal, what I have tries includes calling daemon from nix package and do that in the rust main or setup hook of the builder, however, it seems freeze the tauri UI.

Thanks for your help.

**Comment by FabianLars** [comment]
I don't think a process can detach itself. So it probably comes down to the stuff described here: <https://superuser.com/questions/178587/how-do-i-detach-a-process-from-terminal-entirely>

I do wonder if you can detach the app by restarting it with a Command tho 🤔 I could imagine that it would loose sudo priveleges

**Reply by unixapple** [reply]
Thanks a lot for your reply, after some research, it turns out that the shared library (built from go code which contains a go runtime) which 
cause the daemonize not worked as expected.

Here is a explanation from go git repo:

https://github.com/golang/go/issues/15556

**Comment by unixapple** [comment, marked as answer]
Based what I have researched and tested, for a stand alone  tauri app, actually we can using rust daemonize library to daemonize the program, if you using some shared library built from go, you actually can not do daemonize the tauri app :(
